site2/website/blog/2020-06-18-Apache-Pulsar-2-6-0.md

```diff
@@ -0,0 +1,400 @@
+
+This PR adds the method `deleteForcefully` to support force deleting subscriptions. 
+
+For more information about implementation details, see [PR-6383](https://github.com/apache/pulsar/pull/6383).
+
+# Functions
+
+### Built-in functions
+
+This PR implements the possibility of creating built-in functions in the same way as adding built-in connectors.
+
+For more information about implementation details, see [PR-6895](https://github.com/apache/pulsar/pull/6895).
+
+### Add Go Function heartbeat (and gRPC service) for production usage
+ 
+For more information about implementation details, see [PR-6031](https://github.com/apache/pulsar/pull/6031).
+
+### Add custom property options to functions
+
+This PR allows users to set custom system properties while submitting functions. This can be used to pass credentials via a system property.
+
+For more information about implementation details, see [PR-6348](https://github.com/apache/pulsar/pull/6348).
+
+### Separate TLS configurations of function worker and broker
+ 
+For more information about implementation details, see [PR-6602](https://github.com/apache/pulsar/pull/6602).
+
+### Add the ability to build consumers in functions and sources
+
+Previously, function and source context give their writers an ability to create publishers but not consumers. This PR fixes this issue.
+
+For more information about implementation details, see [PR-6954](https://github.com/apache/pulsar/pull/6954).
+
+# Pulsar SQL
+
+### Support KeyValue schema
+
+Previously, Pulsar SQL could not read the KeyValue schema data.
+
+This PR adds KeyValue schema support for Pulsar SQL. It adds the prefix `key.` for the key field name and `value.` for the value field name.
+
+For more information about implementation details, see [PR-6325](https://github.com/apache/pulsar/pull/6325).
+
+### Support multiple Avro schema versions
+
+Previously, if you have multiple Avro schema versions for a topic, use Pulsar SQL to query data from this topic will introduce some problems. With this change, You can evolve the schema of the topic and keep transitive backward compatibility of all schemas of the topic if you want to query data from this topic. 
+  
+For more information about implementation details, see [PR-4847](https://github.com/apache/pulsar/pull/4847).
+
+# Java client
+
+### Support waiting for inflight messages while closing a producer
+
+Previously, when you closed a producer, pulsar-client immediately failed inflight messages even if they persisted successfully at the broker. Most of the time, users want to wait for those inflight messages rather than fail them. While pulsar-client lib did not provide a way to wait for inflight messages before closing the producer. This PR supports closing API with the flag where you can control waiting for inflight messages. With this change, you can close a producer by waiting for inflight messages and pulsar-client does not fail those messages immediately.
```

Comment:
```suggestion
Previously, when you closed a producer, the pulsar-client immediately failed inflight messages even if it persisted successfully at the broker. Most of the time, users want to wait for those inflight messages rather than fail them. While the pulsar-client library did not provide a way to wait for inflight messages before closing the producer. This PR supports closing API with a flag where you can control waiting for inflight messages. With this change, you can close a producer by waiting for inflight messages and the pulsar-client does not fail those messages immediately.
```